src/components/GenreSelect.js

```diff
@@ -0,0 +1,19 @@
+import {useState} from "react";
+
+export function GenreSelect({genres, selectedGenre, onGenreSelectCb}) {
+    const [selectedGenreState, setSelectedGenreState] = useState(selectedGenre);
+
+    function onGenreSelect(e) {
+        setSelectedGenreState(e.target.value);
+        onGenreSelectCb(e);
+    }
```

Comment:
`onSelect` callback should be passed to the component as a property. It should take a name of a selected genre as an argument, not an event object.